src/sections/layer_and_mask_information_section/mod.rs

```diff
@@ -263,11 +271,22 @@ fn read_layer_record(cursor: &mut PsdCursor) -> Result<LayerRecord, Error> {
     // until we stop seeing them.
     while cursor.peek_4()? == SIGNATURE_EIGHT_BIM || cursor.peek_4()? == SIGNATURE_EIGHT_B64 {
         let _signature = cursor.read_4()?;
-        let _key = cursor.read_4()?;
+        let _key = Vec::from(cursor.read_4()?);
         let additional_layer_info_len = cursor.read_u32()?;
-        cursor.read(additional_layer_info_len)?;
+
+        if _key.as_slice() == KEY_UNICODE_LAYER_NAME {
+            let length = cursor.read_u32()?;
```

Comment:
Is this length unused? If so mind renaming to `_length`

Reply:
Length is not unused:
```rust
&u8_slice_to_u16(label_name).as_slice()[..length as usize]
                                         -------
```
It cuts two null-byte at the end of string.